internal/declarative/v2/reconciler.go

```diff
@@ -622,19 +624,26 @@ func (r *Reconciler) configClient(ctx context.Context, obj Object) (Client, erro
 	return clnt, nil
 }
 
-func (r *Reconciler) ssaStatus(ctx context.Context, obj client.Object,
-	requeueReason metrics.ManifestRequeueReason,
+func (r *Reconciler) ssaStatusIfDiffExist(ctx context.Context, obj Object,
+	requeueReason metrics.ManifestRequeueReason, currentObjStatus shared.Status,
```

Comment:
Naming: I'd say that the "current" status is defined in the `obj.GetStatus()` and the `currentObjStatus` represents "previous", already existing status.
Thus I'd vote for renaming the parameter to `previousObjStatus`

Reply:
yeah makes sense I think it would make it less confusing